core-library/src/main/java/org/silverpeas/core/index/search/SimpleSearchEngine.java

```diff
@@ -108,4 +120,100 @@ private boolean isSpellingNeeded(List<MatchingIndexEntry> results) {
+      if (!isMatchingIndexEntryAvailable(result, userId, authorization)) {
+        continue;
+      }
+      results.add(result);
+    }
+
+    authorization.disableCache();
+
+    return results;
+  }
+
+  private boolean isMatchingIndexEntryAvailable(MatchingIndexEntry mie, String userId,
+      ComponentAuthorization authorization) {
+    // Do not filter and check external components
+    if (enableExternalSearch && isExternalComponent(mie.getServerName())) {
+      mie.setExternalResult(true);
+      // Filter only Publication and Node data
+      String objectType = mie.getObjectType();
+      if ("Versioning".equals(objectType) || "Publication".equals(objectType)
+          || "Node".equals(objectType)) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+
+    String componentId = mie.getComponent();
+    if (componentId.startsWith("kmelia")) {
+      return authorization
+          .isObjectAvailable(componentId, userId, mie.getObjectId(), mie.getObjectType());
+    }
+    // contrôle des droits sur les espaces et les composants
+    String objectType = mie.getObjectType();
+    if ("Space".equals(objectType)) {
+      // check if space is allowed to current user
+      try {
+        return Administration.get().isSpaceAvailable(mie.getObjectId(), userId);
+      } catch (Exception ignored) {
+        return false;
+      }
+    } else if ("Component".equals(objectType)) {
+      // check if component is allowed to current user
+      try {
+        return Administration.get().isComponentAvailable(mie.getObjectId(), userId);
+      } catch (Exception ignored) {
+        return false;
+      }
+    } else if (UserIndexation.OBJECT_TYPE.equals(objectType) &&
+        !DomainProperties.areDomainsVisibleToAll()) {
+      // visibility between domains is limited, check found user domain against current user domain
+      User userFound = User.getById(mie.getObjectId());
+      if (DomainProperties.areDomainsVisibleOnlyToDefaultOne()) {
+        String currentUserDomainId = User.getById(userId).getDomainId();
+        if ("0".equals(currentUserDomainId)) {
+          // current user of default domain can see all users
+          return true;
+        } else {
+          // current user of other domains can see only users of his domain
+          return userFound.getDomainId().equals(currentUserDomainId);
+        }
+      } else if (DomainProperties.areDomainsNonVisibleToOthers()) {
+        // user found must be in same domain of current user
+        String currentUserDomainId = User.getById(userId).getDomainId();
+        return userFound.getDomainId().equals(currentUserDomainId);
+      }
+    }
+    return true;
+  }
+
+  private boolean isExternalComponent(String serverName) {
+    if (StringUtil.isDefined(localServerName) && !localServerName.equalsIgnoreCase(serverName)) {
+      return true;
+    }
+    return false;
+  }
+
+  private ComponentAuthorization getSecurityIntf() throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)